ctapipe/io/pointing.py

```diff
@@ -0,0 +1,131 @@
+        help="If true, raises an exception when trying to extrapolate out of the given table",
+    ).tag(config=True)
+
+    extrapolate = traits.Bool(
+        help="If bounds_error is False, this flag will specify whether values outside"
+        "the available values are filled with nan (False) or extrapolated (True).",
+        default_value=False,
+    ).tag(config=True)
+
+    def __init__(self, h5file=None, **kwargs):
+        super().__init__(**kwargs)
+
+        if h5file is not None and not isinstance(h5file, tables.File):
+            raise TypeError("h5file must be a tables.File")
+        self.h5file = h5file
+
+        self.interp_options: dict[str, Any] = dict(assume_sorted=True, copy=False)
+        if self.bounds_error:
+            self.interp_options["bounds_error"] = True
+        elif self.extrapolate:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = "extrapolate"
+        else:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = np.nan
+
+        self._alt_interpolators = {}
+        self._az_interpolators = {}
+
+    def add_table(self, tel_id, pointing_table):
+        """
+        Add a table to this interpolator
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id
+        pointing_table : astropy.table.Table
+            Table of pointing values, expected columns
+            are ``time`` as ``Time`` column, ``azimuth`` and ``altitude``
+            as quantity columns.
+        """
+        missing = {"time", "azimuth", "altitude"} - set(pointing_table.colnames)
+        if len(missing) > 0:
+            raise ValueError(f"Table is missing required column(s): {missing}")
+
+        if not isinstance(pointing_table["time"], Time):
+            raise TypeError("'time' column of pointing table must be astropy.time.Time")
+
+        for col in ("azimuth", "altitude"):
+            unit = pointing_table[col].unit
+            if unit is None or not u.rad.is_equivalent(unit):
+                raise ValueError(f"{col} must have units compatible with 'rad'")
+
+        # sort first, so it's not done twice for each interpolator
+        pointing_table.sort("time")
+        # interpolate in mjd TAI. Float64 mjd is precise enough for pointing
+        # and TAI is contiguous, so no issues with leap seconds.
+        mjd = pointing_table["time"].tai.mjd
+
+        az = pointing_table["azimuth"].quantity.to_value(u.rad)
+        # prepare azimuth for interpolation by "unwrapping": i.e. turning
+        # [359, 1] into [359, 361]. This assumes that if we get values like
+        # [359, 1] the telescope moved 2 degrees through 0, not 358 degrees
+        # the other way around. This should be true for all telescopes given
+        # the sampling speed of pointing values and their maximum movement speed.
+        # No telescope can turn more than 180° in 2 seconds.
```

Comment:
I think the more relevant remark is that during tracking of a source the telescopes only move quickly close to the poles, and the worst case for those motions are a straight overhead track that flip the Azimuth 180 degrees. But maybe I misunderstand this comment.

I expect how this works in practice depends on how the telescopes handle the very high zenith tracks, do they stop data taking as they swing around?

Reply:
I think there is a telescope requirement that we do not observe very close to a poll (though I have to check exactly how far is allowed). So probably it's not a major issue, at least for CTA. For HESS, for example CT5 can in principle move to altitudes greater than 90°, but in practice due to the effects it has on pointing accuracy, it's no longer used (and the fact that the HESS software also doesn't do this transform correctly).  

It may be worth adding some of this comment's text to the docstring of the class, however, as it is important to remember and should show up in the documentation.

Reply:
> I think the more relevant remark is that during tracking of a source the telescopes only move quickly close to the poles, 

What is implemented and tested here is really the normal case of moving, not related to any movement at the pole, given the sequence of azimuth values and assuming  you interpolate exactly in between, there are 2 solutions for the sequence:
```
[357, 359, 1, 3]
```

Interpolating naïvely:

```
[358, 180, 2]
```

or with the numpy function applied above, the sequence is first turned into:
```
[357, 359, 361, 363]
```
which is then interpolated and wrapped to:

```
[358, 0, 2]
```